tools/integration-test/src/tests/interchain_security/ica_ordered_channel.rs

```diff
@@ -0,0 +1,219 @@
+
+        chains.node_a.chain_driver().local_transfer_token(
+            &chains.node_a.wallets().user1(),
+            &ica_address.as_ref(),
+            &stake_denom.with_amount(ica_fund).as_ref(),
+        )?;
+
+        chains.node_a.chain_driver().assert_eventual_wallet_amount(
+            &ica_address.as_ref(),
+            &stake_denom.with_amount(ica_fund).as_ref(),
+        )?;
+
+        let amount = 1200;
+
+        let msg = MsgSend {
+            from_address: ica_address.to_string(),
+            to_address: chains.node_a.wallets().user2().address().to_string(),
+            amount: vec![Coin {
+                denom: stake_denom.to_string(),
+                amount: Amount(U256::from(amount)),
+            }],
+        };
+
+        let raw_msg = msg.to_any();
+
+        let cosmos_tx = CosmosTx {
+            messages: vec![raw_msg],
+        };
+
+        let raw_cosmos_tx = cosmos_tx.to_any();
+
+        let interchain_account_packet_data = InterchainAccountPacketData::new(raw_cosmos_tx.value);
+
+        let signer = Signer::from_str(&wallet.address().to_string()).unwrap();
+
+        sleep(Duration::from_secs(5));
+
+        relayer.with_supervisor(|| {
+            sleep(Duration::from_secs(1));
+
+            let ica_events = interchain_send_tx(
+                chains.handle_b(),
+                &signer,
+                &channel.connection.connection_id_b.0,
+                interchain_account_packet_data.clone(),
+                Timestamp::from_nanoseconds(120000000000).unwrap(),
+            )?;
+
+            info!("First ICA transfer made with supervisor: {ica_events:#?}");
+
+            sleep(Duration::from_secs(5));
+
+            Ok(())
+        })?;
+
+        sleep(Duration::from_secs(1));
+
+        let ica_events = interchain_send_tx(
+            chains.handle_b(),
+            &signer,
+            &channel.connection.connection_id_b.0,
+            interchain_account_packet_data.clone(),
+            Timestamp::from_nanoseconds(120000000000).unwrap(),
+        )?;
+
+        info!("Second ICA transfer made without supervisor: {ica_events:#?}");
+
+        sleep(Duration::from_secs(5));
+
+        relayer.with_supervisor(|| {
+            sleep(Duration::from_secs(1));
+
+            let ica_events = interchain_send_tx(
+                chains.handle_b(),
+                &signer,
+                &channel.connection.connection_id_b.0,
+                interchain_account_packet_data,
+                Timestamp::from_nanoseconds(120000000000).unwrap(),
+            )?;
+
+            info!("Third ICA transfer made with supervisor: {ica_events:#?}");
+
+            sleep(Duration::from_secs(5));
+
+            Ok(())
+        })?;
+
+        // Check that the ICA account's balance has been debited the sent amount.
+        chains.node_a.chain_driver().assert_eventual_wallet_amount(
+            &ica_address.as_ref(),
+            &stake_denom.with_amount(ica_fund - 3 * amount).as_ref(),
+        )?;
```

Comment:
I think we should move this in the supervisor block to insure that the clearing is done, and maybe add an assertion that the amount is correctly received